packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -141,28 +141,43 @@ String ensureGradle() {
   return gradle;
 }
 
-/// Create android/local.properties if needed.
-File ensureLocalProperties() {
-  final File localProperties = fs.file('android/local.properties');
-  if (!localProperties.existsSync()) {
-    localProperties.writeAsStringSync(
-        'sdk.dir=${escapePath(androidSdk.directory)}\n'
-        'flutter.sdk=${escapePath(Cache.flutterRoot)}\n'
-    );
+/// Create android/local.properties if needed, and update Flutter settings.
+void updateLocalProperties({String projectPath, String buildMode}) {
+  final File localProperties = (projectPath == null)
+      ? fs.file('android/local.properties')
+      : fs.file(fs.path.join(projectPath, 'android', 'local.properties'));
+  bool changed = false;
+
+  SettingsFile settings;
+  if (localProperties.existsSync()) {
+    settings = new SettingsFile.parseFromFile(localProperties);
+  } else {
+    settings = new SettingsFile();
+    settings.values['sdk.dir'] = escapePath(androidSdk.directory);
+    changed = true;
+  }
+  final String escapedRoot = escapePath(Cache.flutterRoot);
+  if (changed || settings.values['flutter.sdk'] != escapedRoot) {
+    settings.values['flutter.sdk'] = escapedRoot;
+    changed = true;
+  }
+  if (buildMode != null && settings.values['flutter.buildMode'] != buildMode) {
+    settings.values['flutter.buildMode']  = buildMode;
+    changed = true;
+  }
+
+  if (changed) {
+    settings.writeContents(localProperties);
   }
```

Comment:
Hurts every time. It's not a one-line conditional if we have to have two lines, so there should be braces according to the Dart style guide. I really wish we'd stop being special snowflakes, and just adopt dartfmt. End of rant. Fixed.